src/System.IO.Pipelines.Text.Primitives/ReadableBufferExtensions.cs

```diff
@@ -151,31 +151,31 @@ public unsafe static ulong GetUInt64(this ReadableBuffer buffer)
             return value;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public unsafe static string GetAsciiString(this Span<byte> span)
         {
-            if (span.IsEmpty)
-            {
+            var len = span.Length;
+            if (len == 0) {
```

Comment:
It's not faster, but note that later in this routine, Span.Length is accessed several times, and so the new code accesses it once and stores the value in a local. But now I have cached Length and so I may as well use it to check if the span is empty.